crates/turbopack-tests/tests/snapshot.rs

```diff
@@ -262,51 +287,52 @@ async fn walk_asset(
     Ok(())
 }
 
+async fn get_contents(file: AssetContentVc) -> Result<Option<String>> {
+    Ok(match &*file.await? {
+        AssetContent::File(file) => match &*file.await? {
+            FileContent::NotFound => None,
+            FileContent::Content(expected) => Some(trimmed_string(expected.content())),
+        },
+        AssetContent::Redirect { target, link_type } => Some(format!(
+            "Redirect {{ target: {target}, link_type: {:?} }}",
+            link_type
+        )),
+    })
+}
+
 async fn diff(
     path: FileSystemPathVc,
     actual: AssetContentVc,
     expected: AssetContentVc,
 ) -> Result<()> {
     let path_str = &path.await?.path;
 
-    let actual = match &*actual.await? {
-        AssetContent::File(file) => match &*file.await? {
-            FileContent::NotFound => bail!("could not generate {} contents", path_str),
-            FileContent::Content(actual) => trimmed_string(actual.content()),
-        },
-        AssetContent::Redirect { target, link_type } => {
-            format!(
-                "Redirect {{ target: {target}, link_type: {:?} }}",
-                link_type
-            )
-        }
-    };
-    let changeset = match &*expected.await? {
-        AssetContent::File(file) => match &*file.await? {
-            FileContent::NotFound => Changeset::new("", &actual, "\n"),
-            FileContent::Content(expected) => {
-                let expected = trimmed_string(expected.content());
-                Changeset::new(&expected, &actual, "\n")
-            }
-        },
-        AssetContent::Redirect { target, link_type } => Changeset::new(
-            &format!(
-                "Redirect {{ target: {target}, link_type: {:?} }}",
-                link_type
-            ),
-            &actual,
-            "\n",
-        ),
+    let actual = match get_contents(actual).await? {
+        Some(s) => s,
+        None => bail!("could not generate {} contents", path_str),
     };
+    let expected = get_contents(expected).await?;
 
-    if changeset.distance > 0 {
+    if Some(&actual) != expected.as_ref() {
         if *UPDATE {
             let content = File::from(actual).into();
             path.write(content).await?;
             println!("updated contents of {}", path_str);
         } else {
-            println!("{}", print_changeset(&changeset));
-            bail!("contents of {} did not match", path_str);
+            if expected.is_none() {
+                println!("new file {path_str} detected:");
+            } else {
+                println!("contents of {path_str} did not match:");
+            }
+            let expected = expected.unwrap_or_default();
+            let diff = TextDiff::from_lines(&expected, &actual);
+            println!(
+                "{}",
+                diff.unified_diff()
+                    .context_radius(3)
+                    .header("expected", "actual")
+            );
```

Comment:
Perhaps use `eprintln` instead of `println` in the above?

Reply:
Done.